core/modal.c

```diff
@@ -0,0 +1,371 @@
+            ) {
+                _lf_trigger_reaction(_lf_startup_reactions[i], -1);
+            }
+        }
+    }
+
+    // Reset the should_trigger_startup flag for startup reactions that are
+    // going to be triggered in the current tag.
+    for (int i = 0; i < _lf_startup_reactions_size; i++) {
+        if (_lf_startup_reactions[i]->mode != NULL) {
+            if (_lf_mode_is_active(_lf_startup_reactions[i]->mode)) {
+                _lf_startup_reactions[i]->mode->should_trigger_startup = false;
+            }
+        }
+    }
+}
+
+// Linked list element for suspended events in inactive modes
+typedef struct _lf_suspended_event {
+    struct _lf_suspended_event* next;
+    event_t* event;
+} _lf_suspended_event_t;
+_lf_suspended_event_t* _lf_suspended_events_head = NULL; // Start of linked collection of suspended events (managed automatically!)
+int _lf_suspended_events_num = 0; // Number of suspended events (managed automatically!)
+_lf_suspended_event_t* _lf_unsused_suspended_events_head = NULL; // Internal collection of reusable list elements (managed automatically!)
+
+/**
+ * Save the given event as suspended.
+ */
+void _lf_add_suspended_event(event_t* event) {
+    _lf_suspended_event_t* new_suspended_event;
+    if (_lf_unsused_suspended_events_head != NULL) {
+        new_suspended_event = _lf_unsused_suspended_events_head;
+        _lf_unsused_suspended_events_head = _lf_unsused_suspended_events_head->next;
+    } else {
+        new_suspended_event = (_lf_suspended_event_t*) malloc(sizeof(_lf_suspended_event_t));
+    }
+
+    new_suspended_event->event = event;
+    new_suspended_event->next = _lf_suspended_events_head; // prepend
+    _lf_suspended_events_num++;
+
+    _lf_suspended_events_head = new_suspended_event;
+}
+
+/**
+ * Removes the given node from the list of suspended events.
+ * Returns the next element in the list.
+ */
+_lf_suspended_event_t* _lf_remove_suspended_event(_lf_suspended_event_t* event) {
+    _lf_suspended_event_t* next = event->next;
+
+    // Clear content
+    event->event = NULL;
+    event->next = NULL;
+    _lf_suspended_events_num--;
+
+    // Store for recycling
+    if (_lf_unsused_suspended_events_head == NULL) {
+        _lf_unsused_suspended_events_head = event;
+    } else {
+        event->next = _lf_unsused_suspended_events_head;
+        _lf_unsused_suspended_events_head = event;
+    }
+
+    if (_lf_suspended_events_head == event) {
+        _lf_suspended_events_head = next; // Adjust head
+    } else {
+        _lf_suspended_event_t* predecessor = _lf_suspended_events_head;
+        while(predecessor->next != event && predecessor != NULL) {
+                predecessor = predecessor->next;
+        }
+        if (predecessor != NULL) {
+                predecessor->next = next; // Remove from linked list
+        }
+    }
+
+    return next;
+}
+
+/**
+ * Performs transitions in all modal reactors.
```

Comment:
```suggestion
 * Perform transitions in all modal reactors.
```
A bit more of an explanation of what this means would be welcome.